public/app/features/alerting/unified/hooks/AI/useAIRuleFeedback.test.ts

```diff
@@ -0,0 +1,133 @@
+import { act, renderHook } from '@testing-library/react';
+
+import { config } from '@grafana/runtime';
+
+import { testWithFeatureToggles } from '../../test/test-utils';
+
+import { useAIRuleFeedback } from './useAIRuleFeedback';
+
+const mockUseURLSearchParams = jest.fn();
+jest.mock('../useURLSearchParams', () => ({
+  useURLSearchParams: () => mockUseURLSearchParams(),
+}));
+
+describe('useAIRuleFeedback', () => {
+  const mockTracker = jest.fn();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('with feature toggle enabled', () => {
+    testWithFeatureToggles(['alertingAIGenAlertRules']);
+
+    it('should return isFromAI as true when fromAI param is true', () => {
+      mockUseURLSearchParams.mockReturnValue([new URLSearchParams('fromAI=true'), jest.fn()]);
```

Comment:
You can try to use this pattern https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/settings/navigation.test.ts#L20C8-L29C8 instead of mocking `useURLSearchParams`

Reply:
ohh yes! thanks!